lib/ansible/modules/storage/ibm/ibm_sa_domain.py

```diff
@@ -0,0 +1,152 @@
+        from IBM Spectrum Accelerate storage systems"
+
+options:
+    domain:
+        description:
+            - domain name
+        required: true
+    state:
+        description:
+            - The desired state of the domain
+        required: true
+        default: "present"
+        choices: [ "present", "absent" ]
+    ldap_id:
+        description:
+            - ldap id to add to the domain.
+        required: false
+    size:
+        description:
+            - domain size
+        required: false
+    hard_capacity:
+        description:
+            - Hard capacity of the domain.
+        required: false
+    soft_capacity:
+        description:
+            - Soft capacity of the domain.
+        required: false
+    max_cgs:
+        description:
+            - Number of max cgs.
+        required: false
+    max_dms:
+        description:
+            - number of max dms.
+        required: false
+    max_mirrors:
+        description:
+            - number of max_mirrors.
+        required: false
+    max_pools:
+        description:
+            - number of max_pools.
+        required: false
+    max_volumes:
+        description:
+            - number of max_volumes.
+        required: false
+    perf_class:
+        description:
+            - Add the domain to a performance class.
+        required: false
+
+extends_documentation_fragment:
+    - ibm_storage
+
+author:
+    - Tzur Eliyahu (tzure@il.ibm.com)
+'''
+
+EXAMPLES = '''
+- name: Define new domain.
+  ibm_sa_domain:
+    domain: domain_name
+    size: domain_size
+    state: present
+    username: admin
+    password: secret
+    endpoints: hostdev-system
+
+- name: Delete domain.
+  ibm_sa_domain:
+    domain: domain_name
+    state: absent
+    username: admin
+    password: secret
+    endpoints: hostdev-system
+'''
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ibm_sa_utils import execute_pyxcli_command, \
+    connect_ssl, spectrum_accelerate_spec, is_pyxcli_installed
+
+
+def main():
+    argument_spec = spectrum_accelerate_spec()
+    argument_spec.update(
+        dict(
+            state=dict(default='present', choices=['present', 'absent']),
+            domain=dict(required=True),
+            size=dict(),
+            max_dms=dict(),
+            max_cgs=dict(),
+            ldap_id=dict(),
+            max_mirrors=dict(),
+            max_pools=dict(),
+            max_volumes=dict(),
+            perf_class=dict(),
+            hard_capacity=dict(),
+            soft_capacity=dict()
+        )
+    )
+
+    module = AnsibleModule(argument_spec)
+
+    is_pyxcli_installed(module)
+
+    xcli_client = connect_ssl(module)
+    domain = xcli_client.cmd.domain_list(
+        domain=module.params['domain']).as_single_element
+    state = module.params['state']
+
+    state_changed = False
+    if state == 'present' and not domain:
+        state_changed = execute_pyxcli_command(
```

Comment:
`execute_pyxcli_command` always returns True, so this module will always report as `changed`. Is this behavior intended ?

Reply:
Yes. Notice that there are 4 options: 
2 options are - the domain does not exist, and we create it, or it does exist and we delete it. In that case, it will call "execute_pyxcli_command" and will return True. in the other 2 cases - where it exists and domain already exists, or delete and the domain does not exist - we will not call pyxcli. pyxcli is called **only** when a change is needed.

Reply:
OK Cool.